### ISSUE TITLE
Renaming of classes

### ISSUE THREAD

**Issue by FBumann** [flixOpt/flixOpt#18]
- Removing the "c" prefix
- Translating into English
- Removing prefix "Base"
- Limiting Acronyms: TS -> TimeSeries; ME -> ModelingElement
https://docs.google.com/spreadsheets/d/1gnYlAPAb8Io-yboUf2w5bRQMYEXfLPee07AwNuLkc5I/edit?pli=1#gid=2087149538

**Comment by FBumann** [comment]
# Renaming of classes

The renaming aims at making the framework more literal and therefore easier to work with, understand and develop. Verbose names are therefore preferred over unclear but short names.
## Following [PEP 8 – Style Guide for Python Code](https://peps.python.org/pep-0008/#class-names)
- Module:   short, all lowercase name -> No Capital Letters, no _, ...
- Package: short, all lowercase name,  -> No Capital Letters, no _, ...
- Classes:  CapWords convention -> FirstLetterOfEachWortCapitalized
- Type Variables: CapWords
- Function Name: lowercase or lower_case_with_underscores
- Variable Name: lowercase or lower_case_with_underscores
- Function & Method Arguments: lowercase or lower_case_with_underscores
- Method & Instance Variable Names: 
  - lowercase or lower_case_with_underscores
  - _one_leading_underscore for non-public methods/instance variables
  - avoid __double_leading_underscore
- Summing up:
  - module, package:               lowercase
  - classes, type variables:     CapWords
  - everything else:                  lowercase or lower_case_with_underscores

**Comment by FBumann** [comment]
# Applying PEP 8 to flixOpt - General Thoughts
### Package
- lowercase would be necessary, but im nut sure
### Modules
- removing prefix "flix"
- More renaming will go hand in hand with reorganizing the package.
### Classes
- Removing Prefix "c"
- Removing _
- Making the class names more verbose and descriptive
- Removing prefix "Base"
### Functions, Attributes, ...
- sometimes more descriptive names
- switching from CapsWords to words_with_underscore

## Classes - ordered by module
## basicModeling
#### cBase Model -> LinearModel
- Manages Variables and Equations and Objective

#### cVariable -> Variable
- A Single variable

#### cVariable_TS -> VariableTS
- A Collection of Variables for a TimeSeries

#### cBeforeValueSet -> StartValues
- ???

#### cEquation -> Equation
- Puts Variables, Factors and Summands into an Equation

#### cVector -> Summand
- Handles a Linear Summand (Variable*Factor)

#### cSolverLog -> SolverLog
- Parses the solver log of different solvers

## flixBasics
### cArgsClass ->??
### cTS_Vector -> TimeSeries
- Handles Time Series Data and access to it via indexing.
### cTS_collection -> TimeSeriesCollection
- Manages a Collection of TimeSeries

## flixBasicsPublic
### cTSraw -> TimeSeriesRaw
- User Interface to initiate a TimeSeries easily
- _Isn't there a way to include this into the main class TimeSeries instead?_
### cInvestArgs -> InvestArgs
- User Interface to initiate cFeatureInvest
- _Isn't there a way to include this into the main class TimeSeries instead?_

## flixComps
#### cEHK ->ElectricBoiler
#### cKWK -> CHP
#### cCoolingTower -> CoolingUnit
#### cBaseLinearTransformer -> LinearTransformer
#### cHeatPump -> HeatPump
#### cAbwaermeHP -> HeatPump...???
- Heat Pump with ambient Heat Source Modeled
#### cStorage -> Storage
#### cSourceAndSink -> SourceAndSink
#### cSource -> Source
#### cSink -> Sink

## flixStructure -> ??
- flixStructure and flixFeatures need to be renamed and restructured together into multiple modules.
  - cME -> Element
    - .modBox -> .system_model
    - .mod -> .element_model
  - cMEModel -> ElementModel
  - cEnergySystem -> System
    - .modBox -> .system_model
  - cModelBoxOfES -> SystemModel
  - cEffectType -> Effect
  - cEffectTypeList -> EffectCollection
  - cMediumCollection -> MediumCollection
  - cBaseComponent -> Component
  - cBus -> Bus
  - cFlow -> Flow
  - cCalculation -> Calculation
  - cGlobal -> Global
  - cTransportation -> Transportation
  - cIO -> ? NotImplementedYet

## flixFeatures ->??
- flixStructure and flixFeatures need to be renamed and restructured together into multiple modules.

## flixAggregation -> aggregation
- Providing the Time Series Aggregation mechanism.
- Might be good to reorganize it together with Calculation, which provides the modeling types
- 

## flixOptHelperFcts -> helpers

## flixPlotHelperFcts -> plots
- Providing simple plotting of results and/or Configurations of Elements

## flixPostprocessing -> results
- reading and organizing saved results of a solved Model

## flixCompsExperimental
- togehter with flixComps, this should be moved in its own sub-package "components", and divided into multiple modules (transformer, experimental, sources/sinks, ...)

**Comment by FBumann** [comment]
# Applying PEP 8 - Proposal by Model
## basicModeling
#### cBase Model -> LinearModel
- Manages Variables and Equations and Objective

#### cVariable -> Variable
- A Single variable

#### cVariable_TS -> VariableTS
- A Collection of Variables for a TimeSeries

#### cBeforeValueSet -> StartValues
- ???

#### cEquation -> Equation
- Puts Variables, Factors and Summands into an Equation

#### cVector -> Summand
- Handles a Linear Summand (Variable*Factor)

#### cSolverLog -> SolverLog
- Parses the solver log of different solvers

## flixBasics
### cArgsClass ->??
### cTS_Vector -> TimeSeries
- Handles Time Series Data and access to it via indexing.
### cTS_collection -> TimeSeriesCollection
- Manages a Collection of TimeSeries

## flixBasicsPublic
### cTSraw -> TimeSeriesRaw
- User Interface to initiate a TimeSeries easily
- _Isn't there a way to include this into the main class TimeSeries instead?_
### cInvestArgs -> InvestArgs
- User Interface to initiate cFeatureInvest
- _Isn't there a way to include this into the main class TimeSeries instead?_

## flixComps
#### cEHK ->ElectricBoiler
#### cKWK -> CHP
#### cCoolingTower -> CoolingUnit
#### cBaseLinearTransformer -> LinearTransformer
#### cHeatPump -> HeatPump
#### cAbwaermeHP -> HeatPump...???
- Heat Pump with ambient Heat Source Modeled
#### cStorage -> Storage
#### cSourceAndSink -> SourceAndSink
#### cSource -> Source
#### cSink -> Sink

## flixStructure -> ??
- flixStructure and flixFeatures need to be renamed and restructured together into multiple modules.
  - cME -> Element
    - .modBox -> .system_model
    - .mod -> .element_model
  - cMEModel -> ElementModel
  - cEnergySystem -> System
    - .modBox -> .system_model
  - cModelBoxOfES -> SystemModel
  - cEffectType -> Effect
  - cEffectTypeList -> EffectCollection
  - cMediumCollection -> MediumCollection
  - cBaseComponent -> Component
  - cBus -> Bus
  - cFlow -> Flow
  - cCalculation -> Calculation
  - cGlobal -> Global
  - cTransportation -> Transportation
  - cIO -> ? NotImplementedYet

## flixFeatures ->??
- flixStructure and flixFeatures need to be renamed and restructured together into multiple modules.

## flixAggregation -> aggregation
- Providing the Time Series Aggregation mechanism.
- Might be good to reorganize it together with Calculation, which provides the modeling types
- 

## flixOptHelperFcts -> helpers

## flixPlotHelperFcts -> plots
- Providing simple plotting of results and/or Configurations of Elements

## flixPostprocessing -> results
- reading and organizing saved results of a solved Model

## flixCompsExperimental
- togehter with flixComps, this should be moved in its own sub-package "components", and divided into multiple modules (transformer, experimental, sources/sinks, ...)

**Comment by baumbude** [comment]
Great. Thank´s! As we discussed I am fine with almost all renaming suggestions.

Here are the few disagreements:
### attention with attribute names ###
Principle: the attribut of type element_model of an element, has just to be named ".model", not ".element_model".
Maybe this is the case for other renaming suggestions as well -> take care!

###  module name prefix flix... ### 
As I said, I still hesitate on clearing the module prefix, cause the motivation was:
- it is easier to see, which one of opened modules in your ide belong to flixOpt package (especially doing several "jumping into" while debugging. Maybe it is better visualized in other IDEs than Spyder?
- sligthly strengthens the brand "flixOpt"
Are there really some fundamental pros of cutting the prefix?

### cTSRaw ###
cTS_Vector/cTSRaw-> TimeSeries / TimeSeriesRaw
I don´t know how many "TimesSeries" do exist in python packages, so for me a more special name would be nice, e.g. flixTimeSeries

### InvestArgs ###
I am not so sure if "args" is really helping the user. Maybe we should change into "InvestSet", "InvestSetUp", "InvestParams"?

### cAbwaermeHP  ###
Really silly name - anything would be better! :)
But we should discuss the implementation: My plan was to introduce obligatory and optional flows of components, so that we could integrate the heat-source-flow into the standard "HeatPump"-component itself.

**Comment by FBumann** [comment]
Thanks for your response.
Regarding your comments:
## .element_model
.model (type ElementModel) is enough, if .system_model (type SystemModel) stays. It must be clear which "model" is which, otherwise it its confusing.

## module name prefix
From PEP 8: "Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged."
Furthermore, looking at widely used packages ( pandas, numpy, matplotlib, ...), a prefix like "flix" is never used. package and model names are kept short and concise.
`import matplotlib.pyplot` shows where the module "pyplot" comes from. There is no need for the prefix "mat". `import matplotlib.matpyplot`.
Coming from a user perspective, `import flixOpt.postprocessing` instead of `import flixOpt.flixPostprocessing` is shorter and more concise.

## InvestArgs
Good Suggestion. Again coming from different style guides: Do not include Datatypes in the name (Especially if it's not correct). Therefore I would suggest `Investment`, which is clear and concise.

## cAbwaermeHP
An Optional Flow in the HeatPump would be perfect.
In the mean time, ill rename this class, to `AmbientHeatPump`

**Comment by baumbude** [comment]
### .element_model ###
Wenn es die Klassen Mensch, Gesellschaftsvermögen und PersoenlichesVermoegen gibt, dann würde für Objekt FelixBumann das Attribut FelixBumann.Vermoegen auch völlig ausreichend sein. Das ist ja ein bisschen der Witz der objektorientierten Programmierung, dass Attribute primär zum Objekt gehören sollten und Bezüge zu übergeordneten möglichst nur wenn unbedingt nötig als Attribute integriert werden und dann als solche gekennzeichnet werden sollten, also FelixBumann.Gesellschaftsvermögen oder gar FelixBumann.refToGesellschaftsvermoegen oder besser FelixBumann.wahlheimat.vermoegen

### module name prefix ###
überzeugt

### cAbwaermeHP ###
more general: HeatPumpWithSource

**Comment by FBumann** [comment]
# .element_model
Ja, idealerweise. Leider hat cME zwei attribute, .modBox und .mod. Was ist was? kann ja nicht beides .model heißen. Deswegen
-  .system_model und .element_model. 
- alternativ: .system_model und . model
Dieses Dilemma lässt sich evtl auflösen, wenn die Verknüpfungen zwischen System, Element, SystemModel und ElementModel aufgeräumt werden ( sind aktuell sehr kreuz und quer)

**Comment by baumbude** [comment]
Ja, ich meinte die Alternative.
Dr.-Ing. Felix Panitz
Am Hang 2
02689 Wehrsdorf
Tel.: 01577 2962394
E-Mail: ***@***.***

FBumann ***@***.***> schrieb am Sa., 22. Juni 2024, 18:55:

**Comment by FBumann** [comment]
Closed with #53
